python/tvm/relay/qnn/transform.py

```diff
@@ -0,0 +1,33 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name
+
+"""QNN Dialect transformation passes."""
+from __future__ import absolute_import
+
+from . import _transform
+
+def QnnLower():
```

Comment:
To make terminology consistent with LLVM/MLIR let us call it Legalize.

Reply:
https://wiki.aalto.fi/display/t1065450/LLVM+Legalization+and+Lowering

Reply:
Possibly can be handled in https://github.com/dmlc/tvm/pull/3672

Reply:
Not sure if Legalize is the right term here. Legalize seems to target-dependent pass. QNN ops would be lowered to a sequence of Relay ops. Those Relay ops might go through Legalize API if need be.

Reply:
I am also not 100% sure. However, we could view the relay core ops as a target and legalization translate the qnn dialect into the core ops.

Reply:
I think `Lower` maybe is one better name. `Legalize` in LLVM means when we have one `type` (such as `i64` in `x86_32`) or instruction (such as `ISD::OR` of `v4i32` in `x86` backend), we make these types or instruction can be supported in one specific backend. `Lower` means we make high-level concept lower into low level instruction / op combine, i.e. the high-level concept (`qnn` here ) is not supported on **all** target and we should lower it into sequence of low level op. Maybe some low level op is not supported by one specific backend, then we legalize it.